pkg/actions/kv_task_results_iterator.go

```diff
@@ -3,48 +3,54 @@ package actions
 import (
 	"context"
 
-	"google.golang.org/protobuf/proto"
-
 	"github.com/treeverse/lakefs/pkg/kv"
-	"github.com/treeverse/lakefs/pkg/logging"
 )
 
 type KVTaskResultIterator struct {
 	it  kv.PrimaryIterator
-	ctx context.Context
+	err error
 }
 
-func NewKVTaskResultIterator(ctx context.Context, store kv.StoreMessage, repositoryID, runID, after string) *KVTaskResultIterator {
-	key := kv.FormatPath(getTasksPath(repositoryID, runID), after)
-	it, err := store.Scan(ctx, key)
+func NewKVTaskResultIterator(ctx context.Context, store kv.StoreMessage, repositoryID, runID, after string) (*KVTaskResultIterator, error) {
+	prefix := GetTasksPath(repositoryID, runID)
+	if after != "" {
+		after = kv.FormatPath(prefix, after)
```

Comment:
Now that changes the behaviour you had in some different iterator where `after` was the full key. here you treat it as relative.

Reply:
After is a full key only in internal iterators - PrefixIterator
Run/Task iterators get the after from outside, user is not aware of full key - each pkg iterator will need to take the 'after' which will be a know (exposed) identifier and create the full key according to its logic

Reply:
Thanks for adding documentation for it.